server/webapp/WEB-INF/rails.new/app/assets/javascripts/version_updater.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2015 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var VersionUpdater = function (staleVersionInfoUrl, updateServerVersionInfoUrl) {
+    this.staleVersionInfoUrl = staleVersionInfoUrl;
+    this.updateServerVersionInfoUrl = updateServerVersionInfoUrl;
+};
+
+(function ($) {
+    $.extend(VersionUpdater.prototype, {
+        update: function () {
+            if (this.canUpdateVersion())
```

Comment:
@maheshp - can you use 2 spaces for javascript?

We should be using 4 spaces only for java code. Also, single if statements are expected to have a paren —

```javascript
if (foo) {
  bar()
}
```